common/phish_detector.py

```diff
@@ -172,6 +174,9 @@ def prep_features(self, directory):
 
 
     def generate_features(self):
+        if self.use_name_in_from != 0:
+            Detector.USE_NAME = True
```

Comment:
Inside of the ```Detector``` class where, in ```Detector.extract_from```, it chooses whether or not to use the name. Part of this PR was changing all detectors to use that method to extract information from the "From" field so all detectors will use ```Detector.USE_NAME``` to decide whether or not to use the name or the email.